### PR TITLE
MGMT-12875: fail gracefully when konnectivity is missing

### PR DESCRIPTION
**Pull request by slaviered**
## List all the issues related to this PR

- [ ] New Feature 
- [ ] Enhancement 
- [x] Bug fix
- [ ] Tests
- [ ] Documentation
- [ ] CI/CD 

## What environments does this code impact?

- [ ] Automation (CI, tools, etc)
- [ ] Cloud
- [x] Operator Managed Deployments
- [x] None

## How was this code tested?

- [ ] assisted-test-infra environment
- [ ] dev-scripts environment
- [ ] Reviewer's test appreciated
- [x] Waiting for CI to do a full test run
- [ ] Manual (Elaborate on how it was tested)
- [ ] No tests needed

## Checklist

- [ ] Title and description added to both, commit and PR.
- [ ] Relevant issues have been associated (see [CONTRIBUTING] guide)
- [ ] This change does not require a documentation update (docstring, `docs`, README, etc)
- [ ] Does this change include unit-tests (note that code changes require unit-tests)

## Reviewers Checklist

- Are the title and description (in both PR and commit) meaningful and clear?
- Is there a bug required (and linked) for this change?
- Should this PR be backported?

[Kubernetes community documentation]: https://github.com/kubernetes/community/blob/master/contributors/guide/pull-requests.md#commit-message-guidelines
[CONTRIBUTING]: https://github.com/openshift/assisted-service/blob/master/CONTRIBUTING.md

### CODE REVIEW

File: internal/controller/controllers/hypershiftagentserviceconfig_controller.go [hunk 1]
```diff
@@ -778,7 +778,9 @@ func newKonnectivityAgentDeployment(ctx context.Context, log logrus.FieldLogger,
 	//read the existing konnectivity deployment
 	ka := &appsv1.Deployment{}
 	if err := asc.Client.Get(ctx, types.NamespacedName{Name: "konnectivity-agent", Namespace: asc.namespace}, ka); err != nil {
-		log.WithError(err).Errorf("Failed retrieving konnectivity-agend Deployment from namespace %s", asc.namespace)
+		err := pkgerror.Wrap(err, fmt.Sprintf("Failed to retrieve konnectivity-agend Deployment from namespace %s", asc.namespace))
```
Comment (danielerez) [inline internal/controller/controllers/hypershiftagentserviceconfig_controller.go:781]: * need another var for err ("err" shadows declaration)
* s/agend/agent

File: internal/controller/controllers/hypershiftagentserviceconfig_controller_test.go [hunk 2]
```diff
@@ -335,6 +335,21 @@ var _ = Describe("HypershiftAgentServiceConfig reconcile", func() {
...
+		mockSpokeClientCache.EXPECT().Get(gomock.Any()).Return(mockSpokeClient, nil)
+		mockSpokeClient.EXPECT().Get(gomock.Any(), gomock.Any(), gomock.Any()).Return(nil).AnyTimes()
+		mockSpokeClient.EXPECT().Update(gomock.Any(), gomock.Any(), gomock.Any()).Return(nil).AnyTimes()
+		mockSpokeClient.EXPECT().List(gomock.Any(), gomock.Any(), gomock.Any()).Return(nil).AnyTimes()
+		
+		Expect(hr.Client.Delete(ctx, konnectivity)).To(Succeed())
+		_, err := hr.Reconcile(ctx, newHypershiftAgentServiceConfigRequest(hsc))
+		Expect(err).ToNot(BeNil())
+		Expect(err.Error()).To(ContainSubstring(
+			fmt.Sprintf("Failed to retrieve konnectivity-agend Deployment from namespace %s",
```
Comment (danielerez) [inline internal/controller/controllers/hypershiftagentserviceconfig_controller_test.go:348]: s/agend/agent

**Comment by romfreiman** [comment]
Connectivity?

File: internal/controller/controllers/hypershiftagentserviceconfig_controller.go [hunk 3]
```diff
@@ -777,8 +777,10 @@ func newHypershiftWebHookAPIService(ctx context.Context, log logrus.FieldLogger,
 func newKonnectivityAgentDeployment(ctx context.Context, log logrus.FieldLogger, asc ASC) (client.Object, controllerutil.MutateFn, error) {
 	//read the existing konnectivity deployment
 	ka := &appsv1.Deployment{}
-	if err := asc.Client.Get(ctx, types.NamespacedName{Name: "konnectivity-agent", Namespace: asc.namespace}, ka); err != nil {
-		log.WithError(err).Errorf("Failed retrieving konnectivity-agend Deployment from namespace %s", asc.namespace)
+	if e := asc.Client.Get(ctx, types.NamespacedName{Name: "konnectivity-agent", Namespace: asc.namespace}, ka); e != nil {
+		err := pkgerror.Wrap(e, fmt.Sprintf("Failed to retrieve konnectivity-agend Deployment from namespace %s", asc.namespace))
```
Comment (osherdp) [inline internal/controller/controllers/hypershiftagentserviceconfig_controller.go:781]:
```suggestion
		err := pkgerror.Wrap(e, fmt.Sprintf("Failed to retrieve konnectivity-agent Deployment from namespace %s", asc.namespace))
```

File: internal/controller/controllers/hypershiftagentserviceconfig_controller_test.go [hunk 4]
```diff
@@ -335,6 +335,21 @@ var _ = Describe("HypershiftAgentServiceConfig reconcile", func() {
...
+		mockSpokeClientCache.EXPECT().Get(gomock.Any()).Return(mockSpokeClient, nil)
+		mockSpokeClient.EXPECT().Get(gomock.Any(), gomock.Any(), gomock.Any()).Return(nil).AnyTimes()
+		mockSpokeClient.EXPECT().Update(gomock.Any(), gomock.Any(), gomock.Any()).Return(nil).AnyTimes()
+		mockSpokeClient.EXPECT().List(gomock.Any(), gomock.Any(), gomock.Any()).Return(nil).AnyTimes()
+
+		Expect(hr.Client.Delete(ctx, konnectivity)).To(Succeed())
+		_, err := hr.Reconcile(ctx, newHypershiftAgentServiceConfigRequest(hsc))
+		Expect(err).ToNot(BeNil())
+		Expect(err.Error()).To(ContainSubstring(
+			fmt.Sprintf("Failed to retrieve konnectivity-agend Deployment from namespace %s",
```
Comment (osherdp) [inline internal/controller/controllers/hypershiftagentserviceconfig_controller_test.go:348]:
```suggestion
			fmt.Sprintf("Failed to retrieve konnectivity-agent Deployment from namespace %s",
```

**Review by osherdp — CHANGES_REQUESTED** [review]
just some typos

**Comment by danielerez** [comment]
> Connectivity?

It's actually just the naming for the k8s service :)
https://kubernetes.io/docs/tasks/extend-kubernetes/setup-konnectivity/

**Comment by slaviered** [comment]
/override ci/prow/edge-e2e-ai-operator-ztp-hypershift-zero-nodes
/override  ci/prow/edge-e2e-ai-operator-ztp-capi

**Pull request merged**